### PR TITLE
refactor(Android): Handle viewport state more reliably

### PR DESCRIPTION
**Pull request by BrandonTR**
### Summary

_Ticket:_ [🤖 | Handle viewport state more reliably](https://app.asana.com/1/15492006741476/project/1205732265579288/task/1210454674414103?focus=true)

When Molecule was introduced with the new event driven state that seemed like a perfect way to approach this and consolidate map/viewport state management into a view model. I'm sure there's more improvements that could be made but I really need to draw the line somewhere.

Since MapBox configuration is somewhat weird I moved it out of the ViewModel into a new manager. This could probably be rolled back in to the viewmodel somehow but maybe in a future ticket.

SheetRoutes were also moved to shared code and then extended for any compose specific functionality.

~~There's a couple edge cases that need to be fixed before this is ready, as well as tests, and splitting this PR up to be more digestible.~~

android
~~- [ ] All user-facing strings added to strings resource in alphabetical order~~
- [x] Expensive calculations are run in `withContext(Dispatchers.Default)` where possible (ideally in shared code)

### Testing

Added viewmodel tests and adjusted existing test to accommodate changes.

### CODE REVIEW

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/pages/MapAndSheetPage.kt [hunk 1]
```diff
@@ -268,7 +271,7 @@ fun MapAndSheetPage(
         routeId: String,
         context: RouteDetailsContext = RouteDetailsContext.Details,
     ) {
-        mapViewModel.setSelectedVehicle(null)
+        mapViewModel.clearSelectedVehicle()
```
[the remark this answers is not in the document]
Comment (BrandonTR) [reply androidApp/src/main/java/com/mbta/tid/mbta_app/android/pages/MapAndSheetPage.kt:275]: The StopSelected state will replace the VehicleSelected state in the VM on 260 when the nav change event is processed

File: shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/viewModel/MapViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+        data object MapStyleLoaded : Event
+
+        data object ClearSelectedVehicle : Event
+
+        data class LayerManagerInitialized(val layerManager: IMapLayerManager) : Event
+
+        data class LocationPermissionsChanged(val hasPermission: Boolean) : Event
+    }
+
+    sealed class State {
```
[the remark this answers is not in the document]
Comment (BrandonTR) [reply shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/viewModel/MapViewModel.kt:111]: I had wanted to do this and it was a path that I had started to go down but it ended up becoming kind of a mess. I originally wanted to eliminate ViewportProvider entirely and just have things communicate to the map through viewmodel events but it was seeming like we'd need to add an abstraction layer on the MapboxMap view so that we could have a representation of ViewportState in shared code and handle the translation in the view layer. This seemed doable but was a fair bit more effort for something I wasn't sure would end up working favorably. I do think viewport state and maybe even layer manager state would be nice to represent in the view model state.

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/pages/MapAndSheetPage.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (BrandonTR) [reply androidApp/src/main/java/com/mbta/tid/mbta_app/android/pages/MapAndSheetPage.kt:275]: Good point, we don't process/check if nav has changed to route details in the map vm but I think we should. Going to make that change 👍
